rai/client.go

```diff
@@ -885,108 +885,123 @@ func (c *Client) ListOAuthClients() ([]OAuthClient, error) {
+}
+
+func (c *Client) DeleteModelsAsync(
+	database, engine string, models []string,
+) (*TransactionAsyncResult, error) {
+	queries := make([]string, 0)
+	for _, model := range models {
+		queries = append(queries, fmt.Sprintf("def delete:rel:catalog:model[\"%s\"] = rel:catalog:model[\"%s\"]", model, model))
+	}
+
+	return c.ExecuteAsync(database, engine, strings.Join(queries, "\n"), nil, false)
 }
 
 func (c *Client) GetModel(database, engine, model string) (*Model, error) {
-	var result listModelsResponse
-	tx := NewTransaction(c.Region, database, engine, "OPEN")
-	data := tx.Payload(makeListModelsAction())
-	err := c.Post(PathTransaction, tx.QueryArgs(), data, &result)
+	models, err := c.ListModels(database, engine)
 	if err != nil {
 		return nil, err
 	}
-	// assert len(result.Actions) == 1
-	for _, item := range result.Actions[0].Result.Models {
+
+	for _, item := range models {
 		if item.Name == model {
 			return &item, nil
 		}
 	}
+
 	return nil, ErrNotFound
 }
 
 func (c *Client) LoadModel(
 	database, engine, name string, r io.Reader,
-) (*TransactionResult, error) {
+) (*TransactionAsyncResult, error) {
 	return c.LoadModels(database, engine, map[string]io.Reader{name: r})
 }
 
 func (c *Client) LoadModels(
 	database, engine string, models map[string]io.Reader,
-) (*TransactionResult, error) {
-	var result TransactionResult
-	tx := Transaction{
-		Region:   c.Region,
-		Database: database,
-		Engine:   engine,
-		Mode:     "OPEN",
-		Readonly: false}
-	actions := []DbAction{}
+) (*TransactionAsyncResult, error) {
+	queries := make([]string, 0)
 	for name, r := range models {
 		model, err := ioutil.ReadAll(r)
 		if err != nil {
 			return nil, err
 		}
-		action := makeLoadModelAction(name, string(model))
-		actions = append(actions, action)
+
+		queries = append(queries, fmt.Sprintf("def insert:rel:catalog:model[\"%s\"] = \"\"\"%s\"\"\"", name, model))
 	}
-	data := tx.Payload(actions...)
-	err := c.Post(PathTransaction, tx.QueryArgs(), data, &result)
-	if err != nil {
-		return nil, err
+
+	return c.Execute(database, engine, strings.Join(queries, "\n"), nil, false)
+}
+
+func (c *Client) LoadModelsAsync(
+	database, engine string, models map[string]io.Reader,
+) (*TransactionAsyncResult, error) {
+	queries := make([]string, 0)
+	for name, r := range models {
+		model, err := ioutil.ReadAll(r)
+		if err != nil {
+			return nil, err
+		}
+
+		queries = append(queries, fmt.Sprintf("def insert:rel:catalog:model[\"%s\"] = \"\"\"%s\"\"\"", name, model))
 	}
-	return &result, nil
+
+	return c.ExecuteAsync(database, engine, strings.Join(queries, "\n"), nil, false)
 }
 
 // Returns a list of model names for the given database.
 func (c *Client) ListModelNames(database, engine string) ([]string, error) {
-	var models listModelsResponse
-	tx := NewTransaction(c.Region, database, engine, "OPEN")
-	data := tx.Payload(makeListModelsAction())
-	err := c.Post(PathTransaction, tx.QueryArgs(), data, &models)
+	modelNames := make([]string, 0)
```

Comment:
Similar issue here. We shouldn't query out the models and all their source code just to return the names. We can write a query to only return the names.